// stream-chat-react-native-core/lib/module/components/Attachment/GalleryImage.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MemoizedGalleryImage = exports.GalleryImageWithContext = exports.GalleryImage = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _ChatContext = require("../../contexts/chatContext/ChatContext");
var _utils = require("../../utils/utils");
var _excluded = ["ImageComponent", "uri"];
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/Attachment/GalleryImage.tsx";
var GalleryImageWithContext = function GalleryImageWithContext(props) {
  var _props$ImageComponent = props.ImageComponent,
    ImageComponent = _props$ImageComponent === void 0 ? _reactNative.Image : _props$ImageComponent,
    uri = props.uri,
    rest = (0, _objectWithoutProperties2["default"])(props, _excluded);
  if (!(0, _utils.isLocalUrl)(uri)) {
    return _react["default"].createElement(ImageComponent, (0, _extends2["default"])({}, rest, {
      accessibilityLabel: "Gallery Image",
      source: {
        uri: (0, _utils.makeImageCompatibleUrl)(uri)
      },
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 20,
        columnNumber: 7
      }
    }));
  }
  return _react["default"].createElement(_reactNative.Image, (0, _extends2["default"])({}, rest, {
    accessibilityLabel: "Gallery Image",
    source: {
      uri: (0, _utils.makeImageCompatibleUrl)(uri)
    },
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 31,
      columnNumber: 5
    }
  }));
};
exports.GalleryImageWithContext = GalleryImageWithContext;
var MemoizedGalleryImage = _react["default"].memo(GalleryImageWithContext, function (prevProps, nextProps) {
  return (0, _utils.getUrlWithoutParams)(prevProps.uri) === (0, _utils.getUrlWithoutParams)(nextProps.uri);
});
exports.MemoizedGalleryImage = MemoizedGalleryImage;
var GalleryImage = function GalleryImage(props) {
  var _useChatContext = (0, _ChatContext.useChatContext)(),
    ImageComponent = _useChatContext.ImageComponent;
  return _react["default"].createElement(MemoizedGalleryImage, (0, _extends2["default"])({
    ImageComponent: ImageComponent
  }, props, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 57,
      columnNumber: 10
    }
  }));
};
exports.GalleryImage = GalleryImage;
