// stream-chat-react-native-core/lib/module/components/MessageInput/components/InputReplyStateHeader.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputReplyStateHeaderWithContext = exports.InputReplyStateHeader = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _MessageInputContext = require("../../../contexts/messageInputContext/MessageInputContext");
var _ThemeContext = require("../../../contexts/themeContext/ThemeContext");
var _TranslationContext = require("../../../contexts/translationContext/TranslationContext");
var _icons = require("../../../icons");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/MessageInput/components/InputReplyStateHeader.tsx";
var styles = _reactNative.StyleSheet.create({
  replyBoxHeader: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingBottom: 10
  },
  replyBoxHeaderTitle: {
    fontSize: 14,
    fontWeight: 'bold'
  }
});
var InputReplyStateHeaderWithContext = function InputReplyStateHeaderWithContext(_ref) {
  var clearQuotedMessageState = _ref.clearQuotedMessageState,
    disabled = _ref.disabled,
    resetInput = _ref.resetInput;
  var _useTranslationContex = (0, _TranslationContext.useTranslationContext)(),
    t = _useTranslationContex.t;
  var _useTheme = (0, _ThemeContext.useTheme)(),
    _useTheme$theme = _useTheme.theme,
    _useTheme$theme$color = _useTheme$theme.colors,
    black = _useTheme$theme$color.black,
    grey = _useTheme$theme$color.grey,
    grey_gainsboro = _useTheme$theme$color.grey_gainsboro,
    _useTheme$theme$messa = _useTheme$theme.messageInput.editingStateHeader,
    editingBoxHeader = _useTheme$theme$messa.editingBoxHeader,
    editingBoxHeaderTitle = _useTheme$theme$messa.editingBoxHeaderTitle;
  return _react["default"].createElement(_reactNative.View, {
    style: [styles.replyBoxHeader, editingBoxHeader],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 51,
      columnNumber: 5
    }
  }, _react["default"].createElement(_icons.CurveLineLeftUp, {
    pathFill: grey_gainsboro,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 52,
      columnNumber: 7
    }
  }), _react["default"].createElement(_reactNative.Text, {
    style: [styles.replyBoxHeaderTitle, {
      color: black
    }, editingBoxHeaderTitle],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 53,
      columnNumber: 7
    }
  }, t('Reply to Message')), _react["default"].createElement(_reactNative.TouchableOpacity, {
    disabled: disabled,
    onPress: function onPress() {
      resetInput();
      clearQuotedMessageState();
    },
    testID: "close-button",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 56,
      columnNumber: 7
    }
  }, _react["default"].createElement(_icons.CircleClose, {
    pathFill: grey,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 64,
      columnNumber: 9
    }
  })));
};
exports.InputReplyStateHeaderWithContext = InputReplyStateHeaderWithContext;
var areEqual = function areEqual(prevProps, nextProps) {
  var prevDisabled = prevProps.disabled;
  var nextDisabled = nextProps.disabled;
  var disabledEqual = prevDisabled === nextDisabled;
  if (!disabledEqual) return false;
  return true;
};
var MemoizedInputReplyStateHeader = _react["default"].memo(InputReplyStateHeaderWithContext, areEqual);
var InputReplyStateHeader = function InputReplyStateHeader(props) {
  var _useMessageInputConte = (0, _MessageInputContext.useMessageInputContext)(),
    clearQuotedMessageState = _useMessageInputConte.clearQuotedMessageState,
    resetInput = _useMessageInputConte.resetInput;
  return _react["default"].createElement(MemoizedInputReplyStateHeader, (0, _extends2["default"])({
    clearQuotedMessageState: clearQuotedMessageState,
    resetInput: resetInput
  }, props, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 99,
      columnNumber: 10
    }
  }));
};
exports.InputReplyStateHeader = InputReplyStateHeader;
InputReplyStateHeader.displayName = 'ReplyStateHeader{messageInput}';
