// stream-chat-react-native-core/lib/commonjs/components/MessageInput/components/InputEditingStateHeader.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputEditingStateHeaderWithContext = exports.InputEditingStateHeader = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _MessageInputContext = require("../../../contexts/messageInputContext/MessageInputContext");
var _ThemeContext = require("../../../contexts/themeContext/ThemeContext");
var _TranslationContext = require("../../../contexts/translationContext/TranslationContext");
var _icons = require("../../../icons");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/MessageInput/components/InputEditingStateHeader.tsx";
var styles = _reactNative.StyleSheet.create({
  editingBoxHeader: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingBottom: 10
  },
  editingBoxHeaderTitle: {
    fontSize: 14,
    fontWeight: 'bold'
  }
});
var InputEditingStateHeaderWithContext = function InputEditingStateHeaderWithContext(_ref) {
  var clearEditingState = _ref.clearEditingState,
    disabled = _ref.disabled,
    resetInput = _ref.resetInput;
  var _useTranslationContex = (0, _TranslationContext.useTranslationContext)(),
    t = _useTranslationContex.t;
  var _useTheme = (0, _ThemeContext.useTheme)(),
    _useTheme$theme = _useTheme.theme,
    _useTheme$theme$color = _useTheme$theme.colors,
    black = _useTheme$theme$color.black,
    grey = _useTheme$theme$color.grey,
    grey_gainsboro = _useTheme$theme$color.grey_gainsboro,
    _useTheme$theme$messa = _useTheme$theme.messageInput.editingStateHeader,
    editingBoxHeader = _useTheme$theme$messa.editingBoxHeader,
    editingBoxHeaderTitle = _useTheme$theme$messa.editingBoxHeaderTitle;
  return _react["default"].createElement(_reactNative.View, {
    style: [styles.editingBoxHeader, editingBoxHeader],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 51,
      columnNumber: 5
    }
  }, _react["default"].createElement(_icons.Edit, {
    pathFill: grey_gainsboro,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 52,
      columnNumber: 7
    }
  }), _react["default"].createElement(_reactNative.Text, {
    style: [styles.editingBoxHeaderTitle, {
      color: black
    }, editingBoxHeaderTitle],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 53,
      columnNumber: 7
    }
  }, t('Editing Message')), _react["default"].createElement(_reactNative.TouchableOpacity, {
    disabled: disabled,
    onPress: function onPress() {
      resetInput();
      clearEditingState();
    },
    testID: "close-button",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 56,
      columnNumber: 7
    }
  }, _react["default"].createElement(_icons.CircleClose, {
    pathFill: grey,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 64,
      columnNumber: 9
    }
  })));
};
exports.InputEditingStateHeaderWithContext = InputEditingStateHeaderWithContext;
var areEqual = function areEqual(prevProps, nextProps) {
  var prevDisabled = prevProps.disabled;
  var nextDisabled = nextProps.disabled;
  var disabledEqual = prevDisabled === nextDisabled;
  if (!disabledEqual) return false;
  return true;
};
var MemoizedInputEditingStateHeader = _react["default"].memo(InputEditingStateHeaderWithContext, areEqual);
var InputEditingStateHeader = function InputEditingStateHeader(props) {
  var _useMessageInputConte = (0, _MessageInputContext.useMessageInputContext)(),
    clearEditingState = _useMessageInputConte.clearEditingState,
    resetInput = _useMessageInputConte.resetInput;
  return _react["default"].createElement(MemoizedInputEditingStateHeader, (0, _extends2["default"])({
    clearEditingState: clearEditingState,
    resetInput: resetInput
  }, props, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 99,
      columnNumber: 10
    }
  }));
};
exports.InputEditingStateHeader = InputEditingStateHeader;
InputEditingStateHeader.displayName = 'EditingStateHeader{messageInput}';
