// stream-chat-react-native-core/lib/module/components/MessageInput/MessageInput.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageInput = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _react = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _useCountdown2 = require("./hooks/useCountdown");
var _contexts = require("../../contexts");
var _AttachmentPickerContext = require("../../contexts/attachmentPickerContext/AttachmentPickerContext");
var _ChannelContext = require("../../contexts/channelContext/ChannelContext");
var _MessageInputContext = require("../../contexts/messageInputContext/MessageInputContext");
var _MessagesContext = require("../../contexts/messagesContext/MessagesContext");
var _OwnCapabilitiesContext = require("../../contexts/ownCapabilitiesContext/OwnCapabilitiesContext");
var _SuggestionsContext = require("../../contexts/suggestionsContext/SuggestionsContext");
var _ThemeContext = require("../../contexts/themeContext/ThemeContext");
var _ThreadContext = require("../../contexts/threadContext/ThreadContext");
var _TranslationContext = require("../../contexts/translationContext/TranslationContext");
var _AttachmentSelectionBar = require("../AttachmentPicker/components/AttachmentSelectionBar");
var _AutoCompleteInput = require("../AutoCompleteInput/AutoCompleteInput");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/MessageInput/MessageInput.tsx";
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _createForOfIteratorHelperLoose(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (it) return (it = it.call(o)).next.bind(it); if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; return function () { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
var styles = _reactNative.StyleSheet.create({
  attachmentSeparator: {
    borderBottomWidth: 1,
    marginBottom: 10
  },
  autoCompleteInputContainer: {
    alignItems: 'center',
    flexDirection: 'row',
    paddingLeft: 16,
    paddingRight: 16
  },
  composerContainer: {
    alignItems: 'flex-end',
    flexDirection: 'row'
  },
  container: {
    borderTopWidth: 1,
    padding: 10
  },
  inputBoxContainer: {
    borderRadius: 20,
    borderWidth: 1,
    flex: 1
  },
  optionsContainer: {
    flexDirection: 'row',
    paddingBottom: 10,
    paddingRight: 10
  },
  replyContainer: {
    paddingBottom: 12,
    paddingHorizontal: 8
  },
  sendButtonContainer: {
    paddingBottom: 10,
    paddingLeft: 10
  },
  suggestionsListContainer: {
    borderRadius: 10,
    elevation: 3,
    left: 8,
    position: 'absolute',
    right: 8,
    shadowOffset: {
      height: 1,
      width: 0
    },
    shadowOpacity: 0.15
  }
});
var MessageInputWithContext = function MessageInputWithContext(props) {
  var additionalTextInputProps = props.additionalTextInputProps,
    asyncIds = props.asyncIds,
    asyncUploads = props.asyncUploads,
    AutoCompleteSuggestionList = props.AutoCompleteSuggestionList,
    closeAttachmentPicker = props.closeAttachmentPicker,
    cooldownEndsAt = props.cooldownEndsAt,
    CooldownTimer = props.CooldownTimer,
    disabled = props.disabled,
    editing = props.editing,
    FileUploadPreview = props.FileUploadPreview,
    fileUploads = props.fileUploads,
    giphyActive = props.giphyActive,
    ImageUploadPreview = props.ImageUploadPreview,
    imageUploads = props.imageUploads,
    Input = props.Input,
    inputBoxRef = props.inputBoxRef,
    InputButtons = props.InputButtons,
    InputEditingStateHeader = props.InputEditingStateHeader,
    InputGiphySearch = props.InputGiphySearch,
    InputReplyStateHeader = props.InputReplyStateHeader,
    isOnline = props.isOnline,
    isValidMessage = props.isValidMessage,
    maxNumberOfFiles = props.maxNumberOfFiles,
    members = props.members,
    mentionedUsers = props.mentionedUsers,
    numberOfUploads = props.numberOfUploads,
    quotedMessage = props.quotedMessage,
    removeFile = props.removeFile,
    removeImage = props.removeImage,
    Reply = props.Reply,
    resetInput = props.resetInput,
    SendButton = props.SendButton,
    sending = props.sending,
    sendMessageAsync = props.sendMessageAsync,
    ShowThreadMessageInChannelButton = props.ShowThreadMessageInChannelButton,
    suggestions = props.suggestions,
    thread = props.thread,
    threadList = props.threadList,
    triggerType = props.triggerType,
    uploadNewFile = props.uploadNewFile,
    uploadNewImage = props.uploadNewImage,
    watchers = props.watchers;
  var _useState = (0, _react.useState)(0),
    _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
    height = _useState2[0],
    setHeight = _useState2[1];
  var _useTheme = (0, _ThemeContext.useTheme)(),
    _useTheme$theme = _useTheme.theme,
    _useTheme$theme$color = _useTheme$theme.colors,
    border = _useTheme$theme$color.border,
    grey_whisper = _useTheme$theme$color.grey_whisper,
    white = _useTheme$theme$color.white,
    white_smoke = _useTheme$theme$color.white_smoke,
    _useTheme$theme$messa = _useTheme$theme.messageInput,
    attachmentSelectionBar = _useTheme$theme$messa.attachmentSelectionBar,
    autoCompleteInputContainer = _useTheme$theme$messa.autoCompleteInputContainer,
    composerContainer = _useTheme$theme$messa.composerContainer,
    container = _useTheme$theme$messa.container,
    inputBoxContainer = _useTheme$theme$messa.inputBoxContainer,
    optionsContainer = _useTheme$theme$messa.optionsContainer,
    replyContainer = _useTheme$theme$messa.replyContainer,
    sendButtonContainer = _useTheme$theme$messa.sendButtonContainer,
    suggestionListContainer = _useTheme$theme$messa.suggestionsListContainer.container;
  var _useAttachmentPickerC = (0, _AttachmentPickerContext.useAttachmentPickerContext)(),
    attachmentPickerBottomSheetHeight = _useAttachmentPickerC.attachmentPickerBottomSheetHeight,
    attachmentSelectionBarHeight = _useAttachmentPickerC.attachmentSelectionBarHeight,
    bottomInset = _useAttachmentPickerC.bottomInset,
    selectedFiles = _useAttachmentPickerC.selectedFiles,
    selectedImages = _useAttachmentPickerC.selectedImages,
    selectedPicker = _useAttachmentPickerC.selectedPicker,
    setMaxNumberOfFiles = _useAttachmentPickerC.setMaxNumberOfFiles,
    setSelectedFiles = _useAttachmentPickerC.setSelectedFiles,
    setSelectedImages = _useAttachmentPickerC.setSelectedImages;
  var _useCountdown = (0, _useCountdown2.useCountdown)(cooldownEndsAt),
    cooldownRemainingSeconds = _useCountdown.seconds;
  (0, _react.useEffect)(function () {
    setMaxNumberOfFiles(maxNumberOfFiles != null ? maxNumberOfFiles : 10);
    return closeAttachmentPicker;
  }, []);
  var _useState3 = (0, _react.useState)(false),
    _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
    hasResetImages = _useState4[0],
    setHasResetImages = _useState4[1];
  var _useState5 = (0, _react.useState)(false),
    _useState6 = (0, _slicedToArray2["default"])(_useState5, 2),
    hasResetFiles = _useState6[0],
    setHasResetFiles = _useState6[1];
  var selectedImagesLength = hasResetImages ? selectedImages.length : 0;
  var imageUploadsLength = hasResetImages ? imageUploads.length : 0;
  var selectedFilesLength = hasResetFiles ? selectedFiles.length : 0;
  var fileUploadsLength = hasResetFiles ? fileUploads.length : 0;
  var imagesForInput = !!thread && !!threadList || !thread && !threadList;
  (0, _react.useEffect)(function () {
    setSelectedImages([]);
    if (imageUploads.length) {
      imageUploads.forEach(function (image) {
        return removeImage(image.id);
      });
    }
    return function () {
      return setSelectedImages([]);
    };
  }, []);
  (0, _react.useEffect)(function () {
    setSelectedFiles([]);
    if (fileUploads.length) {
      fileUploads.forEach(function (file) {
        return removeFile(file.id);
      });
    }
    return function () {
      return setSelectedFiles([]);
    };
  }, []);
  (0, _react.useEffect)(function () {
    if (hasResetImages === false && imageUploadsLength === 0 && selectedImagesLength === 0) {
      setHasResetImages(true);
    }
  }, [imageUploadsLength, selectedImagesLength]);
  (0, _react.useEffect)(function () {
    if (hasResetFiles === false && fileUploadsLength === 0 && selectedFilesLength === 0) {
      setHasResetFiles(true);
    }
  }, [fileUploadsLength, selectedFilesLength]);
  (0, _react.useEffect)(function () {
    if (imagesForInput === false && imageUploads.length) {
      imageUploads.forEach(function (image) {
        return removeImage(image.id);
      });
    }
  }, [imagesForInput]);
  var MEGA_BYTES_TO_BYTES = 1024 * 1024;
  var MAX_FILE_SIZE_TO_UPLOAD_IN_MB = 100;
  var uploadImagesHandler = function uploadImagesHandler() {
    var imageToUpload = selectedImages.find(function (selectedImage) {
      var uploadedImage = imageUploads.find(function (imageUpload) {
        return imageUpload.file.uri === selectedImage.uri || imageUpload.url === selectedImage.uri;
      });
      return !uploadedImage;
    });
    if (imageToUpload && Number(imageToUpload.fileSize) / MEGA_BYTES_TO_BYTES > MAX_FILE_SIZE_TO_UPLOAD_IN_MB) {
      _reactNative.Alert.alert("Maximum file size upload limit reached, please upload an image below " + MAX_FILE_SIZE_TO_UPLOAD_IN_MB + "MB.");
      setSelectedImages(selectedImages.filter(function (selectedImage) {
        return selectedImage.uri !== imageToUpload.uri;
      }));
    } else {
      if (imageToUpload) uploadNewImage(imageToUpload);
    }
  };
  var removeImagesHandler = function removeImagesHandler() {
    var imagesToRemove = imageUploads.filter(function (imageUpload) {
      return !selectedImages.find(function (selectedImage) {
        return selectedImage.uri === imageUpload.file.uri || selectedImage.uri === imageUpload.url;
      });
    });
    imagesToRemove.forEach(function (image) {
      return removeImage(image.id);
    });
  };
  (0, _react.useEffect)(function () {
    if (imagesForInput) {
      if (selectedImagesLength > imageUploadsLength) {
        uploadImagesHandler();
      } else {
        removeImagesHandler();
      }
    }
  }, [selectedImagesLength]);
  (0, _react.useEffect)(function () {
    if (selectedFilesLength > fileUploadsLength) {
      var fileToUpload = selectedFiles.find(function (selectedFile) {
        var uploadedFile = fileUploads.find(function (fileUpload) {
          return fileUpload.file.uri === selectedFile.uri || fileUpload.url === selectedFile.uri;
        });
        return !uploadedFile;
      });
      if (fileToUpload && Number(fileToUpload.size) / MEGA_BYTES_TO_BYTES > MAX_FILE_SIZE_TO_UPLOAD_IN_MB) {
        _reactNative.Alert.alert("Maximum file size upload limit reached, please upload a file below " + MAX_FILE_SIZE_TO_UPLOAD_IN_MB + "MB.");
        setSelectedFiles(selectedFiles.filter(function (selectedFile) {
          return selectedFile.uri !== fileToUpload.uri;
        }));
      } else {
        if (fileToUpload) uploadNewFile(fileToUpload);
      }
    } else {
      var filesToRemove = fileUploads.filter(function (fileUpload) {
        return !selectedFiles.find(function (selectedFile) {
          return selectedFile.uri === fileUpload.file.uri || selectedFile.uri === fileUpload.url;
        });
      });
      filesToRemove.forEach(function (file) {
        return removeFile(file.id);
      });
    }
  }, [selectedFilesLength]);
  (0, _react.useEffect)(function () {
    if (imagesForInput) {
      if (imageUploadsLength < selectedImagesLength) {
        var updatedSelectedImages = selectedImages.filter(function (selectedImage) {
          var uploadedImage = imageUploads.find(function (imageUpload) {
            return imageUpload.file.uri === selectedImage.uri || imageUpload.url === selectedImage.uri;
          });
          return uploadedImage;
        });
        setSelectedImages(updatedSelectedImages);
      } else if (imageUploadsLength > selectedImagesLength) {
        setSelectedImages(imageUploads.map(function (imageUpload) {
          return {
            height: imageUpload.file.height,
            source: imageUpload.file.source,
            uri: imageUpload.url || imageUpload.file.uri,
            width: imageUpload.file.width
          };
        }).filter(Boolean));
      }
    }
  }, [imageUploadsLength]);
  (0, _react.useEffect)(function () {
    if (fileUploadsLength < selectedFilesLength) {
      var updatedSelectedFiles = selectedFiles.filter(function (selectedFile) {
        var uploadedFile = fileUploads.find(function (fileUpload) {
          return fileUpload.file.uri === selectedFile.uri || fileUpload.url === selectedFile.uri;
        });
        return uploadedFile;
      });
      setSelectedFiles(updatedSelectedFiles);
    } else if (fileUploadsLength > selectedFilesLength) {
      setSelectedFiles(fileUploads.map(function (fileUpload) {
        return {
          duration: fileUpload.file.duration,
          mimeType: fileUpload.file.mimeType,
          name: fileUpload.file.name,
          size: fileUpload.file.size,
          uri: fileUpload.file.uri
        };
      }));
    }
  }, [fileUploadsLength]);
  var editingExists = !!editing;
  (0, _react.useEffect)(function () {
    if (editing && inputBoxRef.current) {
      inputBoxRef.current.focus();
    }
    if (!editing && (giphyActive || fileUploads.length > 0 || mentionedUsers.length > 0 || imageUploads.length > 0 || numberOfUploads > 0)) {
      resetInput();
    }
  }, [editingExists]);
  var asyncIdsString = asyncIds.join();
  var asyncUploadsString = Object.values(asyncUploads).map(function (_ref) {
    var state = _ref.state,
      url = _ref.url;
    return "" + state + url;
  }).join();
  (0, _react.useEffect)(function () {
    if (Object.keys(asyncUploads).length) {
      sending.current = true;
      asyncIds.forEach(function (id) {
        return sendMessageAsync(id);
      });
      sending.current = false;
    }
  }, [asyncIdsString, asyncUploadsString, sendMessageAsync]);
  var getMembers = function getMembers() {
    var result = [];
    if (members && Object.values(members).length) {
      Object.values(members).forEach(function (member) {
        if (member.user) {
          result.push(member.user);
        }
      });
    }
    return result;
  };
  var getUsers = function getUsers() {
    var users = [].concat((0, _toConsumableArray2["default"])(getMembers()), (0, _toConsumableArray2["default"])(getWatchers()));
    var uniqueUsers = {};
    for (var _iterator = _createForOfIteratorHelperLoose(users), _step; !(_step = _iterator()).done;) {
      var user = _step.value;
      if (user && !uniqueUsers[user.id]) {
        uniqueUsers[user.id] = user;
      }
    }
    var usersArray = Object.values(uniqueUsers);
    return usersArray;
  };
  var getWatchers = function getWatchers() {
    var result = [];
    if (watchers && Object.values(watchers).length) {
      result.push.apply(result, (0, _toConsumableArray2["default"])(Object.values(watchers)));
    }
    return result;
  };
  var additionalTextInputContainerProps = (0, _extends2["default"])({
    editable: disabled ? false : undefined
  }, additionalTextInputProps);
  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement(_reactNative.View, {
    onLayout: function onLayout(_ref2) {
      var newHeight = _ref2.nativeEvent.layout.height;
      return setHeight(newHeight);
    },
    style: [styles.container, {
      backgroundColor: white,
      borderColor: border
    }, container],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 511,
      columnNumber: 7
    }
  }, editing && _react["default"].createElement(InputEditingStateHeader, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 519,
      columnNumber: 21
    }
  }), quotedMessage && _react["default"].createElement(InputReplyStateHeader, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 520,
      columnNumber: 27
    }
  }), _react["default"].createElement(_reactNative.View, {
    style: [styles.composerContainer, composerContainer],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 521,
      columnNumber: 9
    }
  }, Input ? _react["default"].createElement(Input, {
    additionalTextInputProps: additionalTextInputContainerProps,
    getUsers: getUsers,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 523,
      columnNumber: 13
    }
  }) : _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement(_reactNative.View, {
    style: [styles.optionsContainer, optionsContainer],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 529,
      columnNumber: 15
    }
  }, InputButtons && _react["default"].createElement(InputButtons, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 530,
      columnNumber: 34
    }
  })), _react["default"].createElement(_reactNative.View, {
    style: [styles.inputBoxContainer, {
      borderColor: grey_whisper,
      paddingVertical: giphyActive ? 8 : 12
    }, inputBoxContainer],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 532,
      columnNumber: 15
    }
  }, (typeof editing !== 'boolean' && (editing == null ? void 0 : editing.quoted_message) || quotedMessage) && _react["default"].createElement(_reactNative.View, {
    style: [styles.replyContainer, replyContainer],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 543,
      columnNumber: 19
    }
  }, _react["default"].createElement(Reply, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 544,
      columnNumber: 21
    }
  })), imageUploads.length ? _react["default"].createElement(ImageUploadPreview, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 547,
      columnNumber: 40
    }
  }) : null, imageUploads.length && fileUploads.length ? _react["default"].createElement(_reactNative.View, {
    style: [styles.attachmentSeparator, {
      borderBottomColor: grey_whisper,
      marginHorizontal: giphyActive ? 8 : 12
    }],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 549,
      columnNumber: 19
    }
  }) : null, fileUploads.length ? _react["default"].createElement(FileUploadPreview, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 559,
      columnNumber: 39
    }
  }) : null, giphyActive ? _react["default"].createElement(InputGiphySearch, {
    disabled: !isOnline,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 561,
      columnNumber: 19
    }
  }) : _react["default"].createElement(_reactNative.View, {
    style: [styles.autoCompleteInputContainer, autoCompleteInputContainer],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 563,
      columnNumber: 19
    }
  }, _react["default"].createElement(_AutoCompleteInput.AutoCompleteInput, {
    additionalTextInputProps: additionalTextInputProps,
    cooldownActive: !!cooldownRemainingSeconds,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 564,
      columnNumber: 21
    }
  }))), _react["default"].createElement(_reactNative.View, {
    style: [styles.sendButtonContainer, sendButtonContainer],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 571,
      columnNumber: 15
    }
  }, cooldownRemainingSeconds ? _react["default"].createElement(CooldownTimer, {
    seconds: cooldownRemainingSeconds,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 573,
      columnNumber: 19
    }
  }) : _react["default"].createElement(SendButton, {
    disabled: disabled || sending.current || !isValidMessage() || giphyActive && !isOnline,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 575,
      columnNumber: 19
    }
  })))), _react["default"].createElement(ShowThreadMessageInChannelButton, {
    threadList: threadList,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 585,
      columnNumber: 9
    }
  })), triggerType && suggestions ? _react["default"].createElement(_reactNative.View, {
    style: [suggestionListContainer, styles.suggestionsListContainer, {
      backgroundColor: white,
      bottom: height
    }],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 589,
      columnNumber: 9
    }
  }, _react["default"].createElement(AutoCompleteSuggestionList, {
    active: !!suggestions,
    data: suggestions.data,
    onSelect: suggestions.onSelect,
    queryText: suggestions.queryText,
    triggerType: triggerType,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 596,
      columnNumber: 11
    }
  })) : null, selectedPicker && _react["default"].createElement(_reactNative.View, {
    style: [{
      backgroundColor: white_smoke,
      height: (attachmentPickerBottomSheetHeight ? attachmentPickerBottomSheetHeight + (attachmentSelectionBarHeight != null ? attachmentSelectionBarHeight : 52) : 360) - bottomInset
    }, attachmentSelectionBar],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 606,
      columnNumber: 9
    }
  }, _react["default"].createElement(_AttachmentSelectionBar.AttachmentSelectionBar, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 618,
      columnNumber: 11
    }
  })));
};
var areEqual = function areEqual(prevProps, nextProps) {
  var prevAdditionalTextInputProps = prevProps.additionalTextInputProps,
    prevAsyncUploads = prevProps.asyncUploads,
    prevDisabled = prevProps.disabled,
    prevEditing = prevProps.editing,
    prevFileUploads = prevProps.fileUploads,
    prevGiphyActive = prevProps.giphyActive,
    prevImageUploads = prevProps.imageUploads,
    prevIsOnline = prevProps.isOnline,
    prevIsValidMessage = prevProps.isValidMessage,
    prevMentionedUsers = prevProps.mentionedUsers,
    prevQuotedMessage = prevProps.quotedMessage,
    prevSending = prevProps.sending,
    prevShowMoreOptions = prevProps.showMoreOptions,
    prevSuggestions = prevProps.suggestions,
    prevT = prevProps.t,
    prevThread = prevProps.thread,
    prevThreadList = prevProps.threadList;
  var nextAdditionalTextInputProps = nextProps.additionalTextInputProps,
    nextAsyncUploads = nextProps.asyncUploads,
    nextDisabled = nextProps.disabled,
    nextEditing = nextProps.editing,
    nextFileUploads = nextProps.fileUploads,
    nextGiphyActive = nextProps.giphyActive,
    nextImageUploads = nextProps.imageUploads,
    nextIsOnline = nextProps.isOnline,
    nextIsValidMessage = nextProps.isValidMessage,
    nextMentionedUsers = nextProps.mentionedUsers,
    nextQuotedMessage = nextProps.quotedMessage,
    nextSending = nextProps.sending,
    nextShowMoreOptions = nextProps.showMoreOptions,
    nextSuggestions = nextProps.suggestions,
    nextT = nextProps.t,
    nextThread = nextProps.thread,
    nextThreadList = nextProps.threadList;
  var tEqual = prevT === nextT;
  if (!tEqual) return false;
  var additionalTextInputPropsEven = prevAdditionalTextInputProps === nextAdditionalTextInputProps;
  if (!additionalTextInputPropsEven) return false;
  var disabledEqual = prevDisabled === nextDisabled;
  if (!disabledEqual) return false;
  var editingEqual = !!prevEditing === !!nextEditing;
  if (!editingEqual) return false;
  var imageUploadsEqual = prevImageUploads.length === nextImageUploads.length;
  if (!imageUploadsEqual) return false;
  var giphyActiveEqual = prevGiphyActive === nextGiphyActive;
  if (!giphyActiveEqual) return false;
  var quotedMessageEqual = !!prevQuotedMessage && !!nextQuotedMessage && typeof prevQuotedMessage !== 'boolean' && typeof nextQuotedMessage !== 'boolean' ? prevQuotedMessage.id === nextQuotedMessage.id : !!prevQuotedMessage === !!nextQuotedMessage;
  if (!quotedMessageEqual) return false;
  var sendingEqual = prevSending.current === nextSending.current;
  if (!sendingEqual) return false;
  var showMoreOptionsEqual = prevShowMoreOptions === nextShowMoreOptions;
  if (!showMoreOptionsEqual) return false;
  var isOnlineEqual = prevIsOnline === nextIsOnline;
  if (!isOnlineEqual) return false;
  var isValidMessageEqual = prevIsValidMessage() === nextIsValidMessage();
  if (!isValidMessageEqual) return false;
  var asyncUploadsEqual = Object.keys(prevAsyncUploads).every(function (key) {
    return prevAsyncUploads[key].state === nextAsyncUploads[key].state && prevAsyncUploads[key].url === nextAsyncUploads[key].url;
  });
  if (!asyncUploadsEqual) return false;
  var fileUploadsEqual = prevFileUploads.length === nextFileUploads.length;
  if (!fileUploadsEqual) return false;
  var mentionedUsersEqual = prevMentionedUsers.length === nextMentionedUsers.length;
  if (!mentionedUsersEqual) return false;
  var suggestionsEqual = !!(prevSuggestions != null && prevSuggestions.data) && !!(nextSuggestions != null && nextSuggestions.data) ? prevSuggestions.data.length === nextSuggestions.data.length && prevSuggestions.data.every(function (_ref3, index) {
    var name = _ref3.name;
    return name === nextSuggestions.data[index].name;
  }) : !!prevSuggestions === !!nextSuggestions;
  if (!suggestionsEqual) return false;
  var threadEqual = (prevThread == null ? void 0 : prevThread.id) === (nextThread == null ? void 0 : nextThread.id) && (prevThread == null ? void 0 : prevThread.text) === (nextThread == null ? void 0 : nextThread.text) && (prevThread == null ? void 0 : prevThread.reply_count) === (nextThread == null ? void 0 : nextThread.reply_count);
  if (!threadEqual) return false;
  var threadListEqual = prevThreadList === nextThreadList;
  if (!threadListEqual) return false;
  return true;
};
var MemoizedMessageInput = _react["default"].memo(MessageInputWithContext, areEqual);
var MessageInput = function MessageInput(props) {
  var _useChatContext = (0, _contexts.useChatContext)(),
    isOnline = _useChatContext.isOnline;
  var ownCapabilities = (0, _OwnCapabilitiesContext.useOwnCapabilitiesContext)();
  var _useChannelContext = (0, _ChannelContext.useChannelContext)(),
    _useChannelContext$di = _useChannelContext.disabled,
    disabled = _useChannelContext$di === void 0 ? false : _useChannelContext$di,
    members = _useChannelContext.members,
    watchers = _useChannelContext.watchers;
  var _useMessageInputConte = (0, _MessageInputContext.useMessageInputContext)(),
    additionalTextInputProps = _useMessageInputConte.additionalTextInputProps,
    asyncIds = _useMessageInputConte.asyncIds,
    asyncUploads = _useMessageInputConte.asyncUploads,
    clearEditingState = _useMessageInputConte.clearEditingState,
    clearQuotedMessageState = _useMessageInputConte.clearQuotedMessageState,
    closeAttachmentPicker = _useMessageInputConte.closeAttachmentPicker,
    cooldownEndsAt = _useMessageInputConte.cooldownEndsAt,
    CooldownTimer = _useMessageInputConte.CooldownTimer,
    editing = _useMessageInputConte.editing,
    FileUploadPreview = _useMessageInputConte.FileUploadPreview,
    fileUploads = _useMessageInputConte.fileUploads,
    giphyActive = _useMessageInputConte.giphyActive,
    ImageUploadPreview = _useMessageInputConte.ImageUploadPreview,
    imageUploads = _useMessageInputConte.imageUploads,
    Input = _useMessageInputConte.Input,
    inputBoxRef = _useMessageInputConte.inputBoxRef,
    InputButtons = _useMessageInputConte.InputButtons,
    InputEditingStateHeader = _useMessageInputConte.InputEditingStateHeader,
    InputGiphySearch = _useMessageInputConte.InputGiphySearch,
    InputReplyStateHeader = _useMessageInputConte.InputReplyStateHeader,
    isValidMessage = _useMessageInputConte.isValidMessage,
    maxNumberOfFiles = _useMessageInputConte.maxNumberOfFiles,
    mentionedUsers = _useMessageInputConte.mentionedUsers,
    numberOfUploads = _useMessageInputConte.numberOfUploads,
    quotedMessage = _useMessageInputConte.quotedMessage,
    removeFile = _useMessageInputConte.removeFile,
    removeImage = _useMessageInputConte.removeImage,
    resetInput = _useMessageInputConte.resetInput,
    SendButton = _useMessageInputConte.SendButton,
    sending = _useMessageInputConte.sending,
    sendMessageAsync = _useMessageInputConte.sendMessageAsync,
    SendMessageDisallowedIndicator = _useMessageInputConte.SendMessageDisallowedIndicator,
    setGiphyActive = _useMessageInputConte.setGiphyActive,
    setShowMoreOptions = _useMessageInputConte.setShowMoreOptions,
    showMoreOptions = _useMessageInputConte.showMoreOptions,
    ShowThreadMessageInChannelButton = _useMessageInputConte.ShowThreadMessageInChannelButton,
    uploadNewFile = _useMessageInputConte.uploadNewFile,
    uploadNewImage = _useMessageInputConte.uploadNewImage;
  var _useMessagesContext = (0, _MessagesContext.useMessagesContext)(),
    Reply = _useMessagesContext.Reply;
  var _useSuggestionsContex = (0, _SuggestionsContext.useSuggestionsContext)(),
    AutoCompleteSuggestionHeader = _useSuggestionsContex.AutoCompleteSuggestionHeader,
    AutoCompleteSuggestionItem = _useSuggestionsContex.AutoCompleteSuggestionItem,
    AutoCompleteSuggestionList = _useSuggestionsContex.AutoCompleteSuggestionList,
    suggestions = _useSuggestionsContex.suggestions,
    triggerType = _useSuggestionsContex.triggerType;
  var _useThreadContext = (0, _ThreadContext.useThreadContext)(),
    thread = _useThreadContext.thread;
  var _useTranslationContex = (0, _TranslationContext.useTranslationContext)(),
    t = _useTranslationContex.t;
  if (!ownCapabilities.sendMessage && SendMessageDisallowedIndicator) {
    return _react["default"].createElement(SendMessageDisallowedIndicator, {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 824,
        columnNumber: 12
      }
    });
  }
  return _react["default"].createElement(MemoizedMessageInput, (0, _extends2["default"])({
    additionalTextInputProps: additionalTextInputProps,
    asyncIds: asyncIds,
    asyncUploads: asyncUploads,
    AutoCompleteSuggestionHeader: AutoCompleteSuggestionHeader,
    AutoCompleteSuggestionItem: AutoCompleteSuggestionItem,
    AutoCompleteSuggestionList: AutoCompleteSuggestionList,
    clearEditingState: clearEditingState,
    clearQuotedMessageState: clearQuotedMessageState,
    closeAttachmentPicker: closeAttachmentPicker,
    cooldownEndsAt: cooldownEndsAt,
    CooldownTimer: CooldownTimer,
    disabled: disabled,
    editing: editing,
    FileUploadPreview: FileUploadPreview,
    fileUploads: fileUploads,
    giphyActive: giphyActive,
    ImageUploadPreview: ImageUploadPreview,
    imageUploads: imageUploads,
    Input: Input,
    inputBoxRef: inputBoxRef,
    InputButtons: InputButtons,
    InputEditingStateHeader: InputEditingStateHeader,
    InputGiphySearch: InputGiphySearch,
    InputReplyStateHeader: InputReplyStateHeader,
    isOnline: isOnline,
    isValidMessage: isValidMessage,
    maxNumberOfFiles: maxNumberOfFiles,
    members: members,
    mentionedUsers: mentionedUsers,
    numberOfUploads: numberOfUploads,
    quotedMessage: quotedMessage,
    removeFile: removeFile,
    removeImage: removeImage,
    Reply: Reply,
    resetInput: resetInput,
    SendButton: SendButton,
    sending: sending,
    sendMessageAsync: sendMessageAsync,
    setGiphyActive: setGiphyActive,
    setShowMoreOptions: setShowMoreOptions,
    showMoreOptions: showMoreOptions,
    ShowThreadMessageInChannelButton: ShowThreadMessageInChannelButton,
    suggestions: suggestions,
    t: t,
    thread: thread,
    triggerType: triggerType,
    uploadNewFile: uploadNewFile,
    uploadNewImage: uploadNewImage,
    watchers: watchers
  }, props, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 828,
      columnNumber: 5
    }
  }));
};
exports.MessageInput = MessageInput;
MessageInput.displayName = 'MessageInput{messageInput}';
