// stream-chat-react-native-core/lib/commonjs/components/AutoCompleteInput/AutoCompleteSuggestionItem.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AutoCompleteSuggestionItem = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _AutoCompleteSuggestionCommandIcon = require("./AutoCompleteSuggestionCommandIcon");
var _ThemeContext = require("../../contexts/themeContext/ThemeContext");
var _AtMentions = require("../../icons/AtMentions");
var _Avatar = require("../Avatar/Avatar");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/AutoCompleteInput/AutoCompleteSuggestionItem.tsx";
var styles = _reactNative.StyleSheet.create({
  args: {
    fontSize: 14
  },
  column: {
    flex: 1,
    justifyContent: 'space-evenly',
    paddingLeft: 8
  },
  container: {
    alignItems: 'center',
    flexDirection: 'row',
    paddingHorizontal: 16,
    paddingVertical: 8
  },
  name: {
    fontSize: 14,
    fontWeight: 'bold',
    paddingBottom: 2
  },
  tag: {
    fontSize: 12,
    fontWeight: '600'
  },
  text: {
    fontSize: 14
  },
  title: {
    fontSize: 14,
    fontWeight: 'bold',
    paddingHorizontal: 8
  }
});
var AutoCompleteSuggestionItemWithContext = function AutoCompleteSuggestionItemWithContext(_ref) {
  var itemProps = _ref.itemProps,
    triggerType = _ref.triggerType;
  var _useTheme = (0, _ThemeContext.useTheme)(),
    _useTheme$theme = _useTheme.theme,
    _useTheme$theme$color = _useTheme$theme.colors,
    accent_blue = _useTheme$theme$color.accent_blue,
    black = _useTheme$theme$color.black,
    grey = _useTheme$theme$color.grey,
    _useTheme$theme$messa = _useTheme$theme.messageInput.suggestions,
    _useTheme$theme$messa2 = _useTheme$theme$messa.command,
    argsStyle = _useTheme$theme$messa2.args,
    commandContainer = _useTheme$theme$messa2.container,
    title = _useTheme$theme$messa2.title,
    _useTheme$theme$messa3 = _useTheme$theme$messa.emoji,
    emojiContainer = _useTheme$theme$messa3.container,
    text = _useTheme$theme$messa3.text,
    _useTheme$theme$messa4 = _useTheme$theme$messa.mention,
    avatarSize = _useTheme$theme$messa4.avatarSize,
    column = _useTheme$theme$messa4.column,
    mentionContainer = _useTheme$theme$messa4.container,
    nameStyle = _useTheme$theme$messa4.name;
  if (triggerType === 'mention') {
    var _ref2 = itemProps,
      id = _ref2.id,
      image = _ref2.image,
      name = _ref2.name,
      online = _ref2.online;
    return _react["default"].createElement(_reactNative.View, {
      style: [styles.container, mentionContainer],
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 80,
        columnNumber: 7
      }
    }, _react["default"].createElement(_Avatar.Avatar, {
      image: image,
      name: name,
      online: online,
      size: avatarSize,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 81,
        columnNumber: 9
      }
    }), _react["default"].createElement(_reactNative.View, {
      style: [styles.column, column],
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 82,
        columnNumber: 9
      }
    }, _react["default"].createElement(_reactNative.Text, {
      style: [styles.name, {
        color: black
      }, nameStyle],
      testID: "mentions-item-name",
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 83,
        columnNumber: 11
      }
    }, name || id)), _react["default"].createElement(_AtMentions.AtMentions, {
      pathFill: accent_blue,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 87,
        columnNumber: 9
      }
    }));
  } else if (triggerType === 'emoji') {
    var _ref3 = itemProps,
      _name = _ref3.name,
      unicode = _ref3.unicode;
    return _react["default"].createElement(_reactNative.View, {
      style: [styles.container, emojiContainer],
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 93,
        columnNumber: 7
      }
    }, _react["default"].createElement(_reactNative.Text, {
      style: [styles.text, {
        color: black
      }, text],
      testID: "emojis-item-unicode",
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 94,
        columnNumber: 9
      }
    }, unicode), _react["default"].createElement(_reactNative.Text, {
      style: [styles.text, {
        color: black
      }, text],
      testID: "emojis-item-name",
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 97,
        columnNumber: 9
      }
    }, " " + _name));
  } else if (triggerType === 'command') {
    var _ref4 = itemProps,
      args = _ref4.args,
      _name2 = _ref4.name;
    return _react["default"].createElement(_reactNative.View, {
      style: [styles.container, commandContainer],
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 105,
        columnNumber: 7
      }
    }, _react["default"].createElement(_AutoCompleteSuggestionCommandIcon.AutoCompleteSuggestionCommandIcon, {
      name: _name2,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 106,
        columnNumber: 9
      }
    }), _react["default"].createElement(_reactNative.Text, {
      style: [styles.title, {
        color: black
      }, title],
      testID: "commands-item-title",
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 107,
        columnNumber: 9
      }
    }, (_name2 || '').replace(/^\w/, function (_char) {
      return _char.toUpperCase();
    })), _react["default"].createElement(_reactNative.Text, {
      style: [styles.args, {
        color: grey
      }, argsStyle],
      testID: "commands-item-args",
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 110,
        columnNumber: 9
      }
    }, "/" + _name2 + " " + args));
  } else {
    return null;
  }
};
var areEqual = function areEqual(prevProps, nextProps) {
  var prevItemProps = prevProps.itemProps,
    prevType = prevProps.triggerType;
  var nextItemProps = nextProps.itemProps,
    nextType = nextProps.triggerType;
  var itemPropsEqual = prevItemProps === nextItemProps;
  if (!itemPropsEqual) return false;
  var typeEqual = prevType === nextType;
  if (!typeEqual) return false;
  return true;
};
var MemoizedAutoCompleteSuggestionItem = _react["default"].memo(AutoCompleteSuggestionItemWithContext, areEqual);
var AutoCompleteSuggestionItem = function AutoCompleteSuggestionItem(props) {
  return _react["default"].createElement(MemoizedAutoCompleteSuggestionItem, (0, _extends2["default"])({}, props, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 146,
      columnNumber: 6
    }
  }));
};
exports.AutoCompleteSuggestionItem = AutoCompleteSuggestionItem;
AutoCompleteSuggestionItem.displayName = 'AutoCompleteSuggestionItem{messageInput{suggestions{Item}}}';
