// stream-chat-react-native-core/lib/commonjs/contexts/activeChannelsRefContext/ActiveChannelsRefContext.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.withActiveChannelsRefContext = exports.useActiveChannelsRefContext = exports.ActiveChannelsProvider = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _getDisplayName = require("../utils/getDisplayName");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/contexts/activeChannelsRefContext/ActiveChannelsRefContext.tsx";
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
var ActiveChannelsContext = _react["default"].createContext({
  current: []
});
var ActiveChannelsProvider = function ActiveChannelsProvider(_ref) {
  var children = _ref.children,
    value = _ref.value;
  return _react["default"].createElement(ActiveChannelsContext.Provider, {
    value: value,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 16,
      columnNumber: 8
    }
  }, children);
};
exports.ActiveChannelsProvider = ActiveChannelsProvider;
var useActiveChannelsRefContext = function useActiveChannelsRefContext() {
  return (0, _react.useContext)(ActiveChannelsContext);
};
exports.useActiveChannelsRefContext = useActiveChannelsRefContext;
var withActiveChannelsRefContext = function withActiveChannelsRefContext(Component) {
  var WithActiveChannelsContextComponent = function WithActiveChannelsContextComponent(props) {
    var ActiveChannelsContext = useActiveChannelsRefContext();
    return _react["default"].createElement(Component, (0, _extends2["default"])({}, props, {
      activeChannels: ActiveChannelsContext,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 27,
        columnNumber: 12
      }
    }));
  };
  WithActiveChannelsContextComponent.displayName = "WithActiveChannelsContext" + (0, _getDisplayName.getDisplayName)(Component);
  return WithActiveChannelsContextComponent;
};
exports.withActiveChannelsRefContext = withActiveChannelsRefContext;
