// stream-chat-react-native-core/lib/module/components/RTLComponents/WritingDirectionAwareText.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WritingDirectionAwareText = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _excluded = ["children", "style"];
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/RTLComponents/WritingDirectionAwareText.tsx";
var styles = _reactNative.StyleSheet.create({
  defaultStyle: {
    writingDirection: _reactNative.I18nManager.isRTL ? 'rtl' : 'ltr'
  }
});
var WritingDirectionAwareText = function WritingDirectionAwareText(props) {
  var children = props.children,
    style = props.style,
    rest = (0, _objectWithoutProperties2["default"])(props, _excluded);
  return _react["default"].createElement(_reactNative.Text, (0, _extends2["default"])({}, rest, {
    style: [style, styles.defaultStyle],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 13,
      columnNumber: 5
    }
  }), children);
};
exports.WritingDirectionAwareText = WritingDirectionAwareText;
