// stream-chat-react-native-core/lib/commonjs/contexts/messageOverlayContext/MessageOverlayContext.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.withMessageOverlayContext = exports.useMessageOverlayContext = exports.MessageOverlayProvider = exports.MessageOverlayContext = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _useResettableState = require("./hooks/useResettableState");
var _defaultBaseContextValue = require("../utils/defaultBaseContextValue");
var _getDisplayName = require("../utils/getDisplayName");
var _isTestEnvironment = require("../utils/isTestEnvironment");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/contexts/messageOverlayContext/MessageOverlayContext.tsx";
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
var MessageOverlayContext = _react["default"].createContext(_defaultBaseContextValue.DEFAULT_BASE_CONTEXT_VALUE);
exports.MessageOverlayContext = MessageOverlayContext;
var MessageOverlayProvider = function MessageOverlayProvider(_ref) {
  var children = _ref.children,
    value = _ref.value;
  var messageOverlayContext = (0, _useResettableState.useResettableState)(value);
  return _react["default"].createElement(MessageOverlayContext.Provider, {
    value: messageOverlayContext,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 97,
      columnNumber: 5
    }
  }, children);
};
exports.MessageOverlayProvider = MessageOverlayProvider;
var useMessageOverlayContext = function useMessageOverlayContext() {
  var contextValue = (0, _react.useContext)(MessageOverlayContext);
  if (contextValue === _defaultBaseContextValue.DEFAULT_BASE_CONTEXT_VALUE && !(0, _isTestEnvironment.isTestEnvironment)()) {
    throw new Error("The useMessageOverlayContext hook was called outside the MessageOverlayContext Provider. Make sure you have configured OverlayProvider component correctly - https://getstream.io/chat/docs/sdk/reactnative/basics/hello_stream_chat/#overlay-provider");
  }
  return contextValue;
};
exports.useMessageOverlayContext = useMessageOverlayContext;
var withMessageOverlayContext = function withMessageOverlayContext(Component) {
  var WithMessageOverlayContextComponent = function WithMessageOverlayContextComponent(props) {
    var messageContext = useMessageOverlayContext();
    return _react["default"].createElement(Component, (0, _extends2["default"])({}, props, messageContext, {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 137,
        columnNumber: 12
      }
    }));
  };
  WithMessageOverlayContextComponent.displayName = "WithMessageOverlayContext" + (0, _getDisplayName.getDisplayName)(Component);
  return WithMessageOverlayContextComponent;
};
exports.withMessageOverlayContext = withMessageOverlayContext;
