// stream-chat-react-native-core/lib/commonjs/components/MessageOverlay/MessageActionListItem.js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageActionListItem = exports.MemoizedMessageActionListItem = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _reactNativeGestureHandler = require("react-native-gesture-handler");
var _reactNativeReanimated = _interopRequireDefault(require("react-native-reanimated"));
var _useMessageActionAnimation = require("./hooks/useMessageActionAnimation");
var _ThemeContext = require("../../contexts/themeContext/ThemeContext");
var _utils = require("../../utils/utils");
var _this = this,
  _jsxFileName = "/home/runner/work/stream-chat-react-native/stream-chat-react-native/package/src/components/MessageOverlay/MessageActionListItem.tsx";
var MessageActionListItemWithContext = function MessageActionListItemWithContext(props) {
  var action = props.action,
    actionType = props.actionType,
    icon = props.icon,
    index = props.index,
    length = props.length,
    title = props.title,
    titleStyle = props.titleStyle;
  var _useTheme = (0, _ThemeContext.useTheme)(),
    _useTheme$theme = _useTheme.theme,
    _useTheme$theme$color = _useTheme$theme.colors,
    black = _useTheme$theme$color.black,
    border = _useTheme$theme$color.border,
    messageActions = _useTheme$theme.overlay.messageActions;
  var _useMessageActionAnim = (0, _useMessageActionAnimation.useMessageActionAnimation)({
      action: action
    }),
    animatedStyle = _useMessageActionAnim.animatedStyle,
    onTap = _useMessageActionAnim.onTap;
  return _react["default"].createElement(_reactNativeGestureHandler.TapGestureHandler, {
    onHandlerStateChange: onTap,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 64,
      columnNumber: 5
    }
  }, _react["default"].createElement(_reactNativeReanimated["default"].View, {
    style: [styles.row, index !== length - 1 ? (0, _extends2["default"])({}, styles.bottomBorder, {
      borderBottomColor: border
    }) : {}, animatedStyle, messageActions.actionContainer],
    testID: actionType + "-list-item",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 65,
      columnNumber: 7
    }
  }, _react["default"].createElement(_reactNative.View, {
    style: messageActions.icon,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 74,
      columnNumber: 9
    }
  }, icon), _react["default"].createElement(_reactNative.Text, {
    style: [styles.titleStyle, messageActions.title, {
      color: black
    }, titleStyle],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 75,
      columnNumber: 9
    }
  }, title)));
};
var messageActionIsEqual = function messageActionIsEqual(prevProps, nextProps) {
  return prevProps.length === nextProps.length;
};
var MemoizedMessageActionListItem = _react["default"].memo(MessageActionListItemWithContext, messageActionIsEqual);
exports.MemoizedMessageActionListItem = MemoizedMessageActionListItem;
var MessageActionListItem = function MessageActionListItem(props) {
  return _react["default"].createElement(MemoizedMessageActionListItem, (0, _extends2["default"])({}, props, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 102,
      columnNumber: 6
    }
  }));
};
exports.MessageActionListItem = MessageActionListItem;
var styles = _reactNative.StyleSheet.create({
  bottomBorder: {
    borderBottomWidth: 1
  },
  container: {
    borderRadius: 16,
    marginTop: 8,
    maxWidth: 275
  },
  row: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    minWidth: (0, _utils.vw)(65),
    paddingHorizontal: 20,
    paddingVertical: 10
  },
  titleStyle: {
    paddingLeft: 20
  }
});
